lib/elixir/src/elixir_parser.yrl

```diff
@@ -25,19 +25,21 @@ Nonterminals
   dot_identifier dot_op_identifier dot_do_identifier
   dot_paren_identifier dot_bracket_identifier
   do_block fn_eoe do_eoe end_eoe block_eoe block_item block_list
+  number
   .
 
 Terminals
   identifier kw_identifier kw_identifier_safe kw_identifier_unsafe bracket_identifier
   paren_identifier do_identifier block_identifier
   fn 'end' aliases
-  number char atom atom_safe atom_unsafe bin_string list_string sigil
+  char atom atom_safe atom_unsafe bin_string list_string sigil
   dot_call_op op_identifier
   comp_op at_op unary_op and_op or_op arrow_op match_op in_op in_match_op
   type_op dual_op add_op mult_op two_op three_op pipe_op stab_op when_op assoc_op
   capture_op rel_op
   'true' 'false' 'nil' 'do' eol ';' ',' '.'
   '(' ')' '[' ']' '{' '}' '<<' '>>' '%{}' '%'
+  binary octal decimal float hex
```

Comment:
I am not a fan of calling this `binary` tbh, it's a very overloaded term in Elixir. What if we go with `binary_int`, `octal_int`, `hexadecimal_int`, `decimal_int`, and `float`? \cc @josevalim